Feat: global block styles

kaeizen opened the pull request.

kaeizen committed:
minor fix for global colors

fix spacing, do not render components if panel is not open

kaeizen committed:
fix: unit attributes gets added on reload

kaeizen committed:
color schemes: do not render components if panel is not open

kaeizen committed:
feat: block styles

kaeizen committed:
fix order of global settings

kaeizen committed:
fix popup for sortable picker

kaeizen committed:
fix some errors

kaeizen committed:
update block attributes on page load

kaeizen committed:
use set, fix reset

DAnn2012 commented on src/plugins/global-settings/typography/index.js:
Hello @mxkae 

Could you include in this PR of yours these changes of mine to this file (#3473 ) from over 3 months ago?

Thanks.

DAnn2012 commented:
Hello @mxkae 

Could you please include in this PR these changes I made to file `src/components/variation-picker/index.js` (#3321 ) about 9 months ago?

Thanks.

Commit:
Merge branch 'develop' into feat/block-styles

# Conflicts:
#	src/plugins/global-settings/color-schemes/index.js
#	src/plugins/global-settings/editor-loader.js

Commit:
style tweaks

bfintal reviewed (changes requested):
Bug encountered:
- Add Card 3 design from the design library, then pick the first inner column, then create a new block style. The block would disappear and then the other columns would grow taller:

https://github.com/user-attachments/assets/22046e6a-9961-4512-894f-f8b5f9d94c6e


Changes:
1. For the free, simplify the things present in the UI:

What we have now:
<img width="279" height="152" alt="Screenshot 2025-07-15 at 6 41 36 PM" src="https://github.com/user-attachments/assets/ef30adcf-0f0c-4956-8b07-1f5437906354" />

Change to:
<img width="274" height="78" alt="Screenshot 2025-07-15 at 6 42 05 PM" src="https://github.com/user-attachments/assets/ace00fae-5921-4ad0-b20b-182199d68e37" />
- Remove "(Modified)"
- Remove reset button

2. I added a button (using blocksy with some blocksy styles), then updated to the PR. My style says "Default (Modified)" with a reset button even though nothing was modified in the block (clicking on the reset button does nothing):

<img width="266" height="82" alt="image" src="https://github.com/user-attachments/assets/16b6f52c-ff49-4a8f-a507-ccea4589ff37" />

3. Change it so that if it's in "Default" block style, it shouldn't ever say "(Modified)" and there shouldn't be any reset button.
4. When clicking the plus button, the name field should be auto-focused so you can type right away.
5. Match the normal style of the normal select drop down. Right now it looks different from the normal drop down. Any reason for this?

What we have:
<img width="239" height="145" alt="image" src="https://github.com/user-attachments/assets/f2327ac9-05f1-45d4-95ce-83f4aa379fbb" />

What we normally use:
<img width="248" height="168" alt="image" src="https://github.com/user-attachments/assets/b12afec2-7c70-484a-b1f4-f13be9f1e128" />

6. Change the alignment of the elements like this. If no reset button, occupy the entire width:
<img width="275" height="91" alt="image" src="https://github.com/user-attachments/assets/095ee349-b1af-4a17-b0cd-c905025f4a8c" />

if there is a reset button, just add margin right to the select drop down:
<img width="274" height="94" alt="image" src="https://github.com/user-attachments/assets/78432353-b0ff-45ca-a758-58eb9d22be09" />

7. The update button feels weird but I can't put my finger on it. it's like it's too wide or has too much gaps

kaeizen committed:
Merge branch 'develop' into feat/block-styles

kaeizen committed:
Merge branch 'develop' into feat/block-styles

kaeizen committed:
change implementation

-remove iterator
-add edit/save css
-use block style generator
-remove attributes when using block style

kaeizen committed:
fix selector using clientId

kaeizen committed:
add data-block-style attr when necessary

kaeizen committed:
fix block error for button

Commit:
remove the block card description

Commit:
Merge branch 'feat/block-styles' of https://github.com/gambitph/Stackable into feat/block-styles

kaeizen committed:
use root attribute name

kaeizen committed:
Merge branch 'feat/block-styles' of https://github.com/gambitph/Stackable into feat/block-styles

kaeizen committed:
Merge branch 'develop' into feat/block-styles

kaeizen committed:
update UI

kaeizen committed:
Merge branch 'develop' into feat/block-styles

kaeizen committed:
add orange indicator and description

kaeizen committed:
fix UI for long block style labels

kaeizen committed:
fix popover on close

kaeizen committed:
fix popover onclose

kaeizen committed:
do nothing if selected block style is clicked

Commit:
UI tweaks

Commit:
Merge branch 'develop' into feat/block-styles

Commit:
UI Tweaks

Commit:
added accessibility for when opening the popover

bfintal reviewed (changes requested):
A few more changes:
- When you open the block style popover by clicking on the button, you should be able to press the up and down arrows to switch between the different block styles (focus moves on the buttons), then when you press enter, the block style selected will be selected.
- Add a focus on the "save" and "update" buttons when tabbing through them.
- Clicking on the default block style when the default block style is currently selected should not apply it. Currently, if you add the design "Call to action 1" from the design library, then select the heading block, you can click on the "default" block style and the text would turn black. Expected behavior is it will not do anything.
- When you open the block styles popover, the save style takes a few moments to show up, it should not do this and should just be there right away

https://github.com/user-attachments/assets/415f252a-e324-45a3-9327-562cf3efdc30

kaeizen committed:
minor fix

- do nothing if block style is already "Default"
- add focus on save/update buttons
- fix save/update buttons styles

Commit:
move check for manage capabilities

kaeizen committed:
revert style

kaeizen committed:
Merge branch 'feat/block-styles' of https://github.com/gambitph/Stackable into feat/block-styles

Commit:
Adjusted description

Commit:
updated notice

kaeizen committed:
update UI, add unlink button, add arrow key navigation

kaeizen committed:
Merge branch 'feat/block-styles' of https://github.com/gambitph/Stackable into feat/block-styles

Commit:
updated space of description

kaeizen committed:
fix hover state styles

kaeizen committed:
Merge branch 'feat/block-styles' of https://github.com/gambitph/Stackable into feat/block-styles

bfintal commented on src/components/block-css/use-block-style-generator.js:
```diff
@@ -30,6 +30,7 @@ export const useBlockCssGenerator = props => {
 			oldText.current = attributes.text
 			return oldCss.current
 		}
+
```
```suggestion
```

bfintal committed:
Update src/components/block-css/use-block-style-generator.js [skip ci]

kaeizen committed:
Merge branch 'develop' into feat/block-styles

# Conflicts:
#	plugin.php
#	src/plugins/global-settings/editor-loader.js
#	src/plugins/global-settings/index.js

kaeizen committed:
fix hover
